bin/core/src/api/write/mod.rs

```diff
@@ -41,6 +41,26 @@ pub struct WriteArgs {
   pub user: User,
 }
 
+/// Helper function to handle resource creation errors with proper HTTP status codes
+pub fn handle_resource_creation_error(e: anyhow::Error) -> serror::Error {
+  let error_msg = e.to_string();
+  
+  // Check if this is a validation error and set appropriate status code
+  if error_msg.contains("Resource with name") && error_msg.contains("already exists") {
+    e.status_code(StatusCode::CONFLICT)
+  } else if error_msg.contains("Must provide unique name for resource") {
+    e.status_code(StatusCode::CONFLICT)
+  } else if error_msg.contains("busy") {
+    e.status_code(StatusCode::CONFLICT)
+  } else if error_msg.contains("Valid ObjectIds cannot be used as names") 
+         || error_msg.contains("Must provide non-empty name") 
+         || error_msg.contains("User does not have permissions") {
+    e.status_code(StatusCode::BAD_REQUEST)
+  } else {
+    e.into()
+  }
+}
```

Comment:
I think this is the right idea, but not the best approach. This error handling should be done inline inside `resource::create` to avoid matching on error strings.

Can make `resource::create(...) -> serror::Result<...>` directly to accomplish this.  You can `.status_code(...)` directly on most Result, similar to anyhow's `.context()`.

Note that you may have to fix other usage of `resource::create` due to change in error type, but you can easily convert the serror::Error to anyhow::Error using `.map_err(|e| e.error)`.